File: src/learning_ros/src/scripts/subscriber.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String

def callback(data):
    # Callback function to process incoming messages here data.data means the data type of data is string
    rospy.loginfo("Received: %s", data.data) 

def subscriber_node():
    # Initialize the ROS node with a unique name (you can change 'subscriber_node' to your desired name)
    rospy.init_node('subscriber_node', anonymous=True)

    # Subscribe to the 'my_topic' topic with the message type 'String'
    rospy.Subscriber('talker', String, callback)

    # Spin to keep the node alive and process incoming messages this works like a while loop
    rospy.spin()

if __name__ == '__main__':
    try:
        subscriber_node()
    except rospy.ROSInterruptException:
        pass

File: src/learning_ros/src/scripts/subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

// Callback function to process incoming messages
void callback(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("Received: %s", msg->data.c_str());
}

int main(int argc, char **argv)
{
    // Initialize the ROS node with a unique name (you can change "subscriber_node" to your desired name)
    ros::init(argc, argv, "subscriber_node");

    // Create a ROS node handle
    ros::NodeHandle nh;

    // Subscribe to the "my_topic" topic with the message type "std_msgs/String" and specify the callback function
    ros::Subscriber sub = nh.subscribe("talker", 10, callback);

    // Spin to keep the node alive and process incoming messages
    ros::spin();

    return 0;
}
